### PR TITLE
fix bug makeDir

### PR DESCRIPTION
**Pull request by ddkwork**

### CODE REVIEW

**Comment by mirzakhany** [comment]
Hi @ddkwork thanks for this PR. it generally looks good I'll test and merge it. But Im not sure how it solved #8, that issue is more as state keeping issue.

**Comment by ddkwork** [comment]
> Hi @ddkwork thanks for this PR. it generally looks good I'll test and merge it. But Im not sure how it solved #8, that issue is more an state keeping issue.

I'm sorry I read the wrong title, but makedir does fail to create a directory on Win11 systems

File: internal/repository/filesystem.go [hunk 1]
```diff
@@ -465,12 +466,20 @@ func CreateConfigDir() (string, error) {
 }
 
 func makeDir(dir string) error {
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		if err := os.Mkdir(dir, 0755); err != nil {
-			return err
+	dir = strings.ReplaceAll(dir, "\\", "/")
```
Comment (mirzakhany) [inline internal/repository/filesystem.go:469]: Lets use file path method to normalise the path. 
```suggestion
	dir = filepath.FromSlash(dir)
```

**Review by mirzakhany — APPROVED** [review]
Thanks for this PR

**Comment by mirzakhany** [comment]
> > Hi @ddkwork thanks for this PR. it generally looks good I'll test and merge it. But Im not sure how it solved #8, that issue is more an state keeping issue.
> 
> I'm sorry I read the wrong title, but makedir does fail to create a directory on Win11 systems

No worries at all, I should write better description on my issues ;)  . I left one note on the Pr after testing it. I'll try to add some unit tests soon.

**Comment by mirzakhany** [comment]
I have applied the change with a bit modification.

`
func makeDir(dir string) error {
	dir = filepath.FromSlash(dir)
	fnMakeDir := func() error { return os.MkdirAll(dir, os.ModePerm) }
	info, err := os.Stat(dir)
	switch {
	case err == nil:
		if info.IsDir() {
			return nil // The directory exists
		} else {
			return fmt.Errorf("path exists but is not a directory: %s", dir)
		}
	case os.IsNotExist(err):
		return fnMakeDir()
	default:
		return err
	}
}
`

**Comment by ddkwork** [comment]
nice work



---Original---
From: "Mohsen ***@***.***&gt;
Date: Thu, Apr 4, 2024 05:52 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mirzakhany/chapar] fix bug makeDir (PR #14)




 
@mirzakhany approved this pull request.
 
Thanks for this PR

**Pull request closed without merging**